apps/meteor/app/api/server/v1/im.ts

```diff
@@ -123,18 +123,35 @@ API.v1.addRoute(
 			if (!roomId) {
 				throw new Meteor.Error('error-room-param-not-provided', 'Body param "roomId" is required');
 			}
-			const canAccess = await canAccessRoomIdAsync(roomId, this.userId);
-			if (!canAccess) {
-				return API.v1.unauthorized();
+
+			let rid: IRoom['_id'] = roomId;
+			let subscription: ISubscription | null = null;
+
+			const roomExists = !!(await Rooms.findOneById(roomId));
+			if (!roomExists) {
+				// if the room doesnt exists, should check if the user has a subscription to it, and if has, close it
```

Comment:
```suggestion
				// even if the room doesn't exist, we should allow the user to close the subscription anyways
```
being picky here, but trying to make our intentions more explicit